examples/download_sample_data.py

```diff
@@ -1,280 +0,0 @@
-    elif output_path.endswith("tar.bz2"):
-        with tarfile.open(output_path, "r:bz2") as tar:
-            tar.extractall()
-        os.remove(output_path)
-    elif output_path.endswith("zip"):
-        with zipfile.ZipFile(output_path, 'r') as zipf:
-            zipf.extractall()
-        os.remove(output_path)
-
-
-def _process_dataset(dataset, output_dir, here):
-    '''Process each download spec in datasets.yml
-
-    Typically each dataset list entry in the yml has
-    "files" and "url" and "title" keys/values to show
-    local files that must be present / extracted from
-    a decompression of contents downloaded from the url.
-
-    If a url endswith '/', then all files given
-    are assumed to be added to the url pattern at the
-    end
-    '''
-    if not path.exists(output_dir):
-        os.makedirs(output_dir)
-
-    with DirectoryContext(output_dir):
-        requires_download = False
-        for f in dataset.get('files', []):
-            if not path.exists(f):
-                requires_download = True
-                break
-
-        if not requires_download:
-            print('Skipping {0}'.format(dataset['title']))
-            return
-        url = dataset['url']
-        title_fmt = dataset['title'] + ' {} of {}'
-        if url.endswith('/'):
-            urls = [url + f for f in dataset['files']]
-            output_paths = [os.path.join(here, 'data', fname)
-                            for fname in dataset['files']]
-
-            unpacked = ['.'.join(output_path.split('.')[:(-2 if output_path.endswith('gz') else -1)]) + '*'
-                        for output_path in output_paths]
-        else:
-            urls = [url]
-            output_paths = [path.split(url)[1]]
-            unpacked = dataset['files']
-            if not isinstance(unpacked, (tuple, list)):
-                unpacked = [unpacked]
-        zipped = zip(urls, output_paths, unpacked)
-        for idx, (url, output_path, unpack) in enumerate(zipped):
-            running_title = title_fmt.format(idx + 1, len(urls))
-            if glob.glob(unpack) or os.path.exists(unpack.replace('*','')):
-                # Skip a file if a similar one is downloaded:
-                # i.e. one that has same name but dif't extension
-                print('Skipping {0}'.format(running_title))
-                continue
-            _url_to_binary_write(url, output_path, running_title)
-            _extract_downloaded_archive(output_path)
-
-
-def main():
-    '''Download each dataset specified by datasets.yml in this directory'''
-    here = path.abspath(path.join(path.split(__file__)[0]))
-    info_file = path.join(here, 'datasets.yml')
-    with open(info_file) as f:
-        info = ordered_load(f.read())
-        for topic, downloads in info.items():
-            output_dir = path.join(here, topic)
-            for d in downloads:
-                _process_dataset(d, output_dir, here)
-
-if __name__ == '__main__':
-    main()
```

Comment:
Maybe it's just me, but I've always thought that github has the colors backwards --- it should be red for *adding* code; huge swaths of deleted code like this should be in a nice, friendly color that encourages people to do more of it!